src/plugins/inputs/join.js

```diff
@@ -0,0 +1,33 @@
+"use strict";
+
+var Chan = require("../../models/chan");
+var _ = require("lodash");
+
+exports.commands = ["join"];
+
+exports.input = function(network, chan, cmd, args) {
+	var client = this;
+	if (args.length !== 0) {
+		var irc = network.irc;
+		irc.join(args[0], args[1]);
+	}
+
+	_.zip(args[0].split(/,/g), args[1].split(/,/g))
+		.map(function(chanPairs) {
+			return new Chan({
+				type: Chan.Type.CHANNEL,
+				name: chanPairs[0],
+				key: chanPairs[1],
+			});
+		})
+		.map(function(channel) {
+			network.channels.push(channel);
+			client.emit("join", {
+				network: network.id,
+				chan: channel
+			});
+		});
+
+	this.save();
```

Comment:
That's normal, but not what I meant, sorry.

This is what your code does:

1. User joins a new channel (`client/js/lounge.js`)
2. `join` input handler is called (`inputs/join.js`)
  1. Saves the channel in the user config
  2. Emits a `join` to the IRC server
3. `join` IRC event handler is called (`irc-events/join.js`)
  1. Saves the channel in the user config
  2. Opens the window on the client

There is a clear duplication of logic in (2.i.) and (3.i.). The user config saving needs to happen in only one of these.
I am of the opinion it should happen in `irc-events/join.js` because that would be the most reliable way: it is only called when the joining is actually coming back from the server and is successful. If you do it in `inputs/join.js`, you possible need to check many things, like valid channel name, etc.
Problem is, in `irc-events/join.js`, you lost access to the key, so in my opinion, remembering keys should be the only logic in `inputs/join.js`, and forwarding the input command to `irc-framework`.